src/idna.rs

```diff
@@ -13,15 +13,11 @@ impl Idna {
     ///
     /// ```
     /// use ada_url::Idna;
-    /// assert_eq!(Idna::unicode("xn--meagefactory-m9a.ca"), "meßagefactory.ca");
+    /// assert_eq!(Idna::unicode("xn--meagefactory-m9a.ca").as_ref(), "meßagefactory.ca");
     /// ```
     #[must_use]
-    pub fn unicode(input: &str) -> &str {
-        unsafe {
-            let out = ffi::ada_idna_to_unicode(input.as_ptr().cast(), input.len());
-            let slice = core::slice::from_raw_parts(out.data.cast(), out.length);
-            core::str::from_utf8_unchecked(slice)
-        }
+    pub fn unicode(input: &str) -> ffi::ada_owned_string {
```

Comment:
Instead of ada_owned_string, can you just return String, and free it inside this function?